lib/ui/views/ib/ib_page_view.dart

```diff
@@ -211,6 +231,15 @@ class _IbPageViewState extends State<IbPageView> {
     );
   }
 
+  Future _onTocListTileTap(String title) async {
+    var slug = IbEngineService.getSlug(
+        title.replaceFirst(RegExp(r'[\da-zA-Z]+. '), ''));

```

Comment:
There are a lot of `RegExp` lying around, wouldn't it be better to specify what each of them do? Can be overwhelming initially.

Also can we unify these somewhere?

Reply:
I will add comments and yes we can unify these but `RegExp` is being used in syntaxes, service and in view. I think we should be unifying only the `RegExp` only in page view. Others seems fine in their place.

Reply:
Done, once check that out.